frontend/packages/metal3-plugin/src/components/baremetal-hosts/host-actions-provider.ts

```diff
@@ -0,0 +1,356 @@
+import { useStopNodeMaintenanceModal } from '../modals/StopNodeMaintenanceModal';
+
+const useDeleteAction = (kindObj: K8sKind, host, status) => {
+  const { t } = useTranslation();
+  const launcher = useOverlay();
+  const hidden = ![
+    HOST_STATUS_UNKNOWN,
+    HOST_STATUS_READY,
+    HOST_STATUS_AVAILABLE,
+    HOST_STATUS_UNMANAGED,
+    ...HOST_ERROR_STATES,
+  ].includes(status.status);
+  const factory = useMemo(
+    () => ({
+      delete: () => ({
+        id: 'delete-host',
+        label: t('metal3-plugin~Delete Bare Metal Host'),
+        cta: () =>
+          launcher(DeleteOverlay, {
+            kind: kindObj,
+            resource: host,
+          }),
+        accessReview: asAccessReview(BareMetalHostModel, host, 'delete'),
+      }),
+    }),
+    [t, kindObj, host, launcher],
+  );
+  const action = useMemo<Action[]>(() => (!hidden ? [factory.delete()] : []), [factory, hidden]);
+  return action;
+};
+
+const useEditAction = (kindObj: K8sKind, host: BareMetalHostKind) => {
+  const { t } = useTranslation();
+  const factory = useMemo(
+    () => ({
+      edit: () => ({
+        id: 'edit-host',
+        label: t('metal3-plugin~Edit Bare Metal Host'),
+        cta: {
+          href: `/k8s/ns/${getNamespace(host)}/${kindObj && referenceForModel(kindObj)}/${getName(
+            host,
+          )}/edit`,
+        },
+      }),
+    }),
+    [host, kindObj, t],
+  );
+  const action = useMemo<Action[]>(() => [factory.edit()], [factory]);
+  return action;
+};
+
+export const useSetNodeMaintenanceAction = (
+  host: BareMetalHostKind,
+  hasNodeMaintenanceCapability,
+  nodeMaintenance,
+  nodeName,
+) => {
+  const { t } = useTranslation();
+  const hidden = !nodeName || !hasNodeMaintenanceCapability || !!nodeMaintenance;
+  const startNodeMaintenanceModalLauncher = useStartNodeMaintenanceModalLauncher({ nodeName });
+  const factory = useMemo(
+    () => ({
+      setNodeMaintenance: () => ({
+        id: 'set-node-maintenance',
+        label: t('metal3-plugin~Start Maintenance'),
+        cta: startNodeMaintenanceModalLauncher,
+        accessReview: host && asAccessReview(NodeMaintenanceModel, host, 'update'),
+      }),
+    }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [host, t],
+  );
+  const action = useMemo<Action[]>(() => (!hidden ? [factory.setNodeMaintenance()] : []), [
+    factory,
+    hidden,
+  ]);
+  return action;
+};
+
+export const useRemoveNodeMaintenanceAction = (
+  host,
+  hasNodeMaintenanceCapability,
+  nodeMaintenance,
+  nodeName,
+  maintenanceModel,
+) => {
+  const { t } = useTranslation();
+  const hidden = !nodeName || !hasNodeMaintenanceCapability || !nodeMaintenance;
+  const stopNodeMaintenanceModalLauncher = useStopNodeMaintenanceModal(nodeMaintenance);
+  const factory = useMemo(
+    () => ({
+      removeNodeMaintenance: () => ({
+        id: 'remove-node-maintenance',
+        label: t('metal3-plugin~Remove Node Maintenance'),
```

Comment:
is this renaming intended? In the modal and button, we see `Stop/Stop Maintenance` shall we be consistent across all places?

<img width="571" height="260" alt="Image" src="https://github.com/user-attachments/assets/662ed704-94dd-4e9a-a4f3-efbba755d69f" />

<img width="643" height="562" alt="Image" src="https://github.com/user-attachments/assets/aadf43ac-7dd9-4dc7-9f72-eaa6acf4e6ad" />